mungegithub/mungers/owner-label.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mungers
+
+import (
+	"k8s.io/kubernetes/pkg/util/sets"
+	"k8s.io/test-infra/mungegithub/features"
+	"k8s.io/test-infra/mungegithub/github"
+	"k8s.io/test-infra/mungegithub/options"
+
+	githubapi "github.com/google/go-github/github"
+)
+
+// OwnerLabelMunger will label issues as specified in OWNERS files.
+type OwnerLabelMunger struct {
+	labeler fileLabeler
+}
+
+type fileLabeler interface {
+	AllPossibleOwnerLabels() sets.String
+	FindLabelsForPath(path string) sets.String
+}
+
+func init() {
+	ownerLabel := &OwnerLabelMunger{}
+	RegisterMungerOrDie(ownerLabel)
+}
+
+// Name is the name usable in --pr-mungers
+func (b *OwnerLabelMunger) Name() string { return "owner-label" }
+
+// RequiredFeatures is a slice of 'features' that must be provided
+func (b *OwnerLabelMunger) RequiredFeatures() []string {
+	return []string{features.RepoFeatureName, features.AliasesFeature}
+}
+
+// Initialize will initialize the munger
+func (b *OwnerLabelMunger) Initialize(config *github.Config, features *features.Features) error {
+	b.labeler = features.Repos
+	return nil
+}
+
+// EachLoop is called at the start of every munge loop
+func (b *OwnerLabelMunger) EachLoop() error { return nil }
+
+// RegisterOptions registers config options for this munger.
+func (b *OwnerLabelMunger) RegisterOptions(opts *options.Options) {}
+
+func (b *OwnerLabelMunger) getLabels(files []*githubapi.CommitFile) sets.String {
+	labels := sets.String{}
+	for _, file := range files {
+		if file == nil {
+			continue
+		}
+		if file.Changes == nil || *file.Changes == 0 {
+			continue
+		}
+		fileLabels := b.labeler.FindLabelsForPath(*file.Filename)
+		labels = labels.Union(fileLabels)
+	}
+	return labels
+}
+
+// Munge is the workhorse the will actually make updates to the PR
+func (b *OwnerLabelMunger) Munge(obj *github.MungeObject) {
```

Comment:
I think it makes more sense to add a second input source to an existing munger than to make another munger that does that same thing with a different input source, but I'm fine with this as it is now too since it makes it easier to use one label source but not the other, and because the OwnerLabelMunger requires additional features.